src/librustc/hir/lowering.rs

```diff
@@ -4058,16 +4058,16 @@ impl<'a> LoweringContext<'a> {
                 // expand <head>
                 let head = self.lower_expr(head);
                 let head_sp = head.span;
+                let desugared_span = self.allow_internal_unstable(
+                    CompilerDesugaringKind::ForLoop,
+                    head.span,
```

Comment:
would it be better if we use the local variable "head_sp" here instead of de-referencing head.span again ?

Reply:
It doesn't really make a difference, but I suppose that would make more sense!